app/src/js/functional/states.ts

```diff
@@ -591,7 +591,7 @@ export function makeDefaultId (): IdType {
 /**
  * Generate new label id. It should not be called outside this file.
  */
-function genLabelId (): IdType {
```

Comment:
`genLabelId`  and `genShapeId` were intentionally not importable because every shape or label should have its own unique id when it is "made". The `id` should accompany that object for the lifetime.

Reply:
Then there will be a clone method to get exported.

Reply:
The `make*` methods actually support clone.

Reply:
Changed. See the new commit.